app/src/main/java/com/cornellappdev/uplift/ui/components/Home/HomeCard.kt

```diff
@@ -0,0 +1,128 @@
+                        ) {
+                            Text(
+                                text =gymexample.name,
+                                fontSize = 20.sp,
+                                fontFamily = montserratFamily
+
+                                )
+                            Spacer(modifier = Modifier.weight(1F))
+                            Image(
+                                painter = painterResource(id = R.drawable.ic_dumbbell),
+                                contentDescription = "Dumbell",
+                            )
+                            Image(
+                                painter = painterResource(id = R.drawable.ic_bowling_pins),
+                                contentDescription = "Bowling Pins",
+                            )
+
+                        }
+                        Row(modifier = Modifier.padding(start = 5.dp, end = 5.dp)) {
+                            Text(
+                                text = "Open",
+                                fontSize = 12.sp,
+                                color = Color.Green,
+                                fontFamily = montserratFamily
+
+                                )
+                            Spacer(modifier = Modifier.width(10.dp))
+                            Text(
+                                text = "Closes at 6 pm",
+                                fontSize = 12.sp,
+                                fontFamily = montserratFamily
+                                )
+
+                        }
+                        Row(
+                            modifier = Modifier.padding(
+                                start = 5.dp,
+                                end = 5.dp
+                            )
+                        ) {
+                            Text(
+                                text = "Cramped",
+                                fontSize = 12.sp,
+                                color = Color.Red,
+                                fontFamily = montserratFamily
+                                )
+                            Spacer(modifier = Modifier.width(10.dp))
+                            Text(
+                                text = " 120/140",
+                                fontSize = 12.sp,
+                                fontFamily = montserratFamily
+                            )
+                            Spacer(modifier = Modifier.weight(1F))
+                            Text(
+                                text="1.2mi",
+                                fontFamily = montserratFamily
+                            )
+
+                        }
+                    }
+                }
+            }
+            /**
+             * Changes Icon image to either be filled star or regular star depending on if the user has liked
+             * that particular gym.
+              */
+            IconButton(
+                onClick = {
+                    gymexample.toggleFavorite() },
```

Comment:
This button seems to have some issues:

1. Its padding is not quite correct according to the Figma.
2. Its colors are off--it's just purely black right now, even when filled. It should be white outlined with a yellow fill.
3. It doesn't seem to actually update when pressed... this is the most confusing issue, and I'll try my best to explain: When you declare `paintres` above, you declare it as a remembered `MutableState` with a default value based on `gymexample.isFavorite()`. As this is a remembered `MutableState`, the value of `gymexample.isFavorite()` is read once, and only once, and therefore never causes recomposition. Then, the value of the state `paintres` never changes to cause recomposition. 

We want to read directly and recompose based off `gymexample.isFavorite()`! You can change this by directly referencing `gymexample.isFavorite()` in your code instead of putting it as the default value of something:

Ex: `var paintres = if (gymexample.isFavorite()) R.drawable.ic_star_filled else R.drawable.ic_star`
(and then, of course, changing `paintres.value` to just `paintres`.)